api/utils/keys/policy_test.go

```diff
@@ -14,85 +14,203 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package keys
+package keys_test
 
 import (
+	"fmt"
 	"testing"
 
 	"github.com/gravitational/trace"
 	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/api/utils/keys"
 )
 
-// TestVerifyPolicy tests VerifyPolicy.
-func TestVerifyPolicy(t *testing.T) {
-	t.Run("key policy none", func(t *testing.T) {
-		require.NoError(t, PrivateKeyPolicyNone.VerifyPolicy(PrivateKeyPolicyNone))
-		require.NoError(t, PrivateKeyPolicyNone.VerifyPolicy(PrivateKeyPolicyHardwareKey))
-		require.NoError(t, PrivateKeyPolicyNone.VerifyPolicy(PrivateKeyPolicyHardwareKeyTouch))
-	})
-	t.Run("key policy hardware_key", func(t *testing.T) {
-		require.Error(t, PrivateKeyPolicyHardwareKey.VerifyPolicy(PrivateKeyPolicyNone))
-		require.NoError(t, PrivateKeyPolicyHardwareKey.VerifyPolicy(PrivateKeyPolicyHardwareKey))
-		require.NoError(t, PrivateKeyPolicyHardwareKey.VerifyPolicy(PrivateKeyPolicyHardwareKeyTouch))
-	})
-	t.Run("key policy hardware_key_touch", func(t *testing.T) {
-		require.Error(t, PrivateKeyPolicyHardwareKeyTouch.VerifyPolicy(PrivateKeyPolicyNone))
-		require.Error(t, PrivateKeyPolicyHardwareKeyTouch.VerifyPolicy(PrivateKeyPolicyHardwareKey))
-		require.NoError(t, PrivateKeyPolicyHardwareKeyTouch.VerifyPolicy(PrivateKeyPolicyHardwareKeyTouch))
-	})
-}
+var (
+	privateKeyPolicies = []keys.PrivateKeyPolicy{
+		keys.PrivateKeyPolicyNone,
+		keys.PrivateKeyPolicyHardwareKey,
+		keys.PrivateKeyPolicyHardwareKeyTouch,
+		keys.PrivateKeyPolicyHardwareKeyPIN,
+		keys.PrivateKeyPolicyHardwareKeyTouchAndPIN,
+	}
+	hardwareKeyPolicies = []keys.PrivateKeyPolicy{
+		keys.PrivateKeyPolicyHardwareKey,
+		keys.PrivateKeyPolicyHardwareKeyTouch,
+		keys.PrivateKeyPolicyHardwareKeyPIN,
+		keys.PrivateKeyPolicyHardwareKeyTouchAndPIN,
+	}
+	hardwareKeyTouchPolicies = []keys.PrivateKeyPolicy{
+		keys.PrivateKeyPolicyHardwareKeyTouch,
+		keys.PrivateKeyPolicyHardwareKeyTouchAndPIN,
+	}
+	hardwareKeyPINPolicies = []keys.PrivateKeyPolicy{
+		keys.PrivateKeyPolicyHardwareKeyPIN,
+		keys.PrivateKeyPolicyHardwareKeyTouchAndPIN,
+	}
+	hardwareKeyTouchAndPINPolicies = []keys.PrivateKeyPolicy{
+		keys.PrivateKeyPolicyHardwareKeyTouchAndPIN,
+	}
+)
 
-// TestPrivateKeyPolicyError tests private key policy error logic.
-func TestPrivateKeyPolicyError(t *testing.T) {
+func TestIsRequiredPolicyMet(t *testing.T) {
+	privateKeyPolicies := []keys.PrivateKeyPolicy{
+		keys.PrivateKeyPolicyNone,
+		keys.PrivateKeyPolicyHardwareKey,
+		keys.PrivateKeyPolicyHardwareKeyTouch,
+		keys.PrivateKeyPolicyHardwareKeyPIN,
+		keys.PrivateKeyPolicyHardwareKeyTouchAndPIN,
+	}
 	for _, tc := range []struct {
-		desc               string
-		errIn              error
-		expectIsKeyPolicy  bool
-		expectKeyPolicyErr bool
-		expectKeyPolicy    PrivateKeyPolicy
+		requiredPolicy   keys.PrivateKeyPolicy
+		verifiedPolicies []keys.PrivateKeyPolicy
```

Comment:
```suggestion
		matchingPolicies []keys.PrivateKeyPolicy
```

Assuming I read it correctly?